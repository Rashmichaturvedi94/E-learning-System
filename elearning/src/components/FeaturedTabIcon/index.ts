export { FeaturedTabIcon } from './FeaturedTabIcon';
export type { FeaturedTabIconProps } from './FeaturedTabIcon.interface';
