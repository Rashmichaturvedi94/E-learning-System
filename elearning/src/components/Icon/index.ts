export { default as HomeIcon } from 'assets/icons/home.svg';
export { default as SearchIcon } from 'assets/icons/search.svg';
export { default as PlayIcon } from 'assets/icons/play.svg';
export { default as FavIcon } from 'assets/icons/heart.svg';
export { default as ProfileIcon } from 'assets/icons/profile.svg';
export { default as Logo } from 'assets/icons/logo.png';
export { default as BadgeIcon } from 'assets/icons/badge1.svg';
