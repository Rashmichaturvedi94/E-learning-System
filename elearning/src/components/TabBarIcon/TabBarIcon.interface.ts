import Svg, { SvgProps } from 'react-native-svg';

export interface TabBarIconProps extends SvgProps {
  focused?: boolean;
  Icon: Svg;
}
