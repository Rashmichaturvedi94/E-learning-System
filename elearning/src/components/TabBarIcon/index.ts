export { TabBarIcon } from './TabBarIcon';
export type { TabBarIconProps } from './TabBarIcon.interface';
