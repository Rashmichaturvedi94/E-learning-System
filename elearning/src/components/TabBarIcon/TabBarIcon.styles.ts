import styled from 'styled-components/native';
import { TabBarIconProps } from './TabBarIcon.interface';

export const TabBarIcon = styled.View<Pick<TabBarIconProps, 'focused'>>`
  flex: 1;
  opacity: ${({ focused }) => (focused ? 1 : 0.6)};
  align-items: center;
  justify-content: center;
`;
