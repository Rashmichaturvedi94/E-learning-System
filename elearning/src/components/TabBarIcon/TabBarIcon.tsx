import React, { FC } from 'react';
import { TabBarIcon as TabBarIconComponent } from './TabBarIcon.styles';
import { TabBarIconProps } from './TabBarIcon.interface';

export const TabBarIcon: FC<TabBarIconProps> = ({
  focused,
  Icon,
  ...iconProps
}) => (
  <TabBarIconComponent focused={focused}>
    <Icon {...iconProps} />
  </TabBarIconComponent>
);
