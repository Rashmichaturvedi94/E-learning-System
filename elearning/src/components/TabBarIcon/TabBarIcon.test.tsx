import React from 'react';
import { render } from '@testing-library/react-native';
import { HomeIcon } from 'components/Icon';
import { TabBarIcon } from './TabBarIcon';

describe('components/TabBarIcon', () => {
  it('should render', () => {
    const component = render(<TabBarIcon Icon={HomeIcon} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
