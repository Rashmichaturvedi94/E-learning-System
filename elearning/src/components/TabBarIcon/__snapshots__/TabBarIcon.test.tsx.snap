// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TabBarIcon should render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "opacity": 0.6,
      },
    ]
  }
>
  <SvgMock />
</View>
`;
